src/panels/lovelace/cards/hui-area-card.ts

```diff
@@ -0,0 +1,418 @@
+      !this._config ||
+      !this.hass ||
+      !this._entitiesDialog ||
+      !this._entitiesToggle
+    ) {
+      return html``;
+    }
+
+    const area = this._getArea(this._config.area, this._areas);
+
+    if (this._config.area && !area) {
+      return html`<hui-warning>Area Not Found!</hui-warning>`; // Lokalise
+    }
+
+    if (
+      this._entitiesDialog.length === 0 &&
+      this._entitiesToggle.length === 0
+    ) {
+      return html`<hui-warning>No Entities in Area!</hui-warning>`; // Lokalise
+    }
+
+    const toggleEntities: EntitiesCardEntityConfig[] = processConfigEntities(
+      this._entitiesToggle!
+    )
+      .map((entityConfig) => ({
+        tap_action: { action: "toggle" },
+        hold_action: { action: "more-info" },
+        ...entityConfig,
+      }))
+      .slice(0, 3) as EntitiesCardEntityConfig[];
+
+    const dialogEntities: EntitiesCardEntityConfig[] = processConfigEntities(
+      this._entitiesDialog!
+    )
+      .map((entityConfig) => ({
+        tap_action: { action: "more-info" },
+        hold_action: { action: "more-info" },
+        ...entityConfig,
+      }))
+      .slice(0, 3) as EntitiesCardEntityConfig[];
+
+    return html`
+      <ha-card>
+        <img src=${this.hass.hassUrl(this._config.image)} />
+        <div class="container">
+          <div class="sensors">
+            ${dialogEntities.map((entityConf) => {
+              const stateObj = this.hass!.states[entityConf.entity];
+              return html`
+                <span>
+                  <ha-icon
+                    .config=${entityConf}
+                    .icon=${stateIcon(stateObj)}
+                    .actionHandler=${actionHandler({
+                      hasHold: hasAction(entityConf.hold_action),
+                    })}
+                    @action=${this._handleAction}
+                  ></ha-icon>
+                  ${computeStateDisplay(
+                    this.hass!.localize,
+                    stateObj,
+                    this.hass!.locale
+                  )}
+                </span>
+              `;
+            })}
+          </div>
+          <div class="name">${area!.name}</div>
+          <div class="buttons">
+            ${toggleEntities.map((entityConf) => {
+              const stateObj = this.hass!.states[entityConf.entity];
+              return html`
+                <ha-icon-button
+                  .config=${entityConf}
+                  .icon=${stateIcon(stateObj)}
+                  class=${classMap({
+                    off: STATES_OFF.includes(stateObj.state),
+                  })}
+                  .actionHandler=${actionHandler({
+                    hasHold: hasAction(entityConf.hold_action),
+                  })}
+                  @action=${this._handleAction}
+                ></ha-icon-button>
+              `;
+            })}
+          </div>
+        </div>
+      </ha-card>
+    `;
+  }
+
+  protected updated(changedProps: PropertyValues): void {
```

Comment:
Most of this logic (if it doesnt use the dom, and sets a property that is used in render) should be moved to `willUpdate`

Reply:
Pushed logic to will update - Noting Paulus's comment above. Does this negate needing Memoize for that logic